aries_cloudagent/protocols/out_of_band/v1_0/messages/invitation.py

```diff
@@ -86,6 +85,26 @@ def aka(self) -> int:
         return self.value.aka
 
 
+class ServiceOrDIDField(fields.Field):
+    """DIDComm Service object or DID string field for Marshmallow."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if isinstance(value, Service):
+            return value.serialize()
+        return value
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if isinstance(value, dict):
+            return Service.deserialize(value)
+        elif isinstance(value, str):
+            if bool(IndyDID.PATTERN.match(value)):
```

Comment:
I don't think this is the right place to be restricting the implementation to only Indy DIDs. Even now we should be able to accept an invitation with multiple DID services as long as one of them is supported.

Reply:
I have created `DIDValidation` pattern [less retrictive] in `messaging/valid.py`.